Marlin/src/HAL/STM32/timers.h

```diff
@@ -31,7 +31,7 @@
 #define FORCE_INLINE __attribute__((always_inline)) inline
 
 #define hal_timer_t uint32_t
-#define HAL_TIMER_TYPE_MAX 0xFFFFFFFF // Timers can be 16 or 32 bit
+#define HAL_TIMER_TYPE_MAX UINT16_MAX // Timers can be 16 or 32-bit. Use 16-bit to ensure compatibility.
```

Comment:
I found this macro that can probably be used to make the appropriate width determination at runtime. This exists in the CMSIS layer for each individual chip, so _should_ always be correct.

It's not immediately clear to me how much of an advantage it will be.  It ends up coercing the timer value in `HAL_timer_start` if the desired pulse is too slow. It's not clear to me how properly that ends up handled by `stepper.cpp` if that happens.

```
/********************** TIM Instances : 32 bit Counter ************************/
#define IS_TIM_32B_COUNTER_INSTANCE(INSTANCE)(((INSTANCE) == TIM2) || \
                                              ((INSTANCE) == TIM5))
```

Reply:
I implemented this, but opted not to use it for now, for two reasons.

1. Almost every board is already using 16-bit step timers. They won't benefit from making this more flexible, unless extra effort is expended to optimize timer assignments for every board.

2. `HAL_TIMER_TYPE_MAX` currently assumes that all timers share the same maximum count. It would have to either be renamed to restrict it to only matching the step timer, or it would need to take an argument to allow different ranges. That would require touching every HAL, which doesn't seem worth it for earlier mentioned reasons.